Cabal/Distribution/Simple/Setup.hs

```diff
@@ -332,6 +332,12 @@ data ConfigFlags = ConfigFlags {
 
     configDistPref :: Flag FilePath, -- ^"dist" prefix
     configCabalFilePath :: Flag FilePath, -- ^ Cabal file to use
+    configBaseDir       :: Flag FilePath, -- ^ The directory containing the
```

Comment:
So now  `takeBaseName configCabalFilePath == configBaseDir` doesn't need to hold? I still don't quite understand why you need a new option and can't use `takeBaseName $ configCabalFilePath`.

Reply:
You are not bound to have the `configCabalFilePath`. E.g. given the following invocation:
```
defaultMainWithHooksNoReadArgs hooks gpd ["--basedir", "/path/to/happiness"]
```
you do not have the `--cabalConfigFilePath`.

And I'm not sure we want to force providing the `--cabalConfigFilePath`? If we can agree that we want to require the `--cabalConfigFilePath` in those cases. I'm happy to change to `takeBaseName`.

Reply:
> You are not bound to have the `configCabalFilePath`.

But this isn't different from the situation when `configBaseDir` is `NoFlag`. Your example can be written as `defaultMainWithHooksNoReadArgs hooks gpd ["--cabal-file", "/path/to/happiness/happiness.cabal"]` with `--cabal-file`.

Reply:
You are right. the more verbose solution, might be the less error prone. Let's do it that way.